### PR TITLE
UniFFI: supports bindings generated from multiple crates

### PR DESCRIPTION
**Pull request by MatMaul**
This changes allow to build a functional wheel of [Matrix Rust SDK](https://github.com/matrix-org/matrix-rust-sdk) FFI bindings.

### CODE REVIEW

**Comment by messense** [comment]
Thanks for the PR, could you please add a test crate in `test-crates/` and write a test case? (See https://github.com/PyO3/maturin/blob/main/tests/run.rs)

File: src/module_writer.rs [hunk 1]
```diff
@@ -1150,12 +1156,16 @@ pub fn write_uniffi_module(
             ))?;
 
             File::create(base_path.join("__init__.py"))?.write_all(py_init.as_bytes())?;
-            let target = base_path.join(&binding_name).with_extension("py");
-            fs::copy(&uniffi_binding, &target).context(format!(
-                "Failed to copy {} to {}",
-                uniffi_binding.display(),
-                target.display()
-            ))?;
+            if let Ok(read_dir) = fs::read_dir(binding_dir.clone()) {
```
Comment (messense) [inline src/module_writer.rs:1159]:
```suggestion
            if let Ok(read_dir) = fs::read_dir(&binding_dir) {
```

File: src/module_writer.rs [hunk 2]
```diff
@@ -1150,12 +1156,16 @@ pub fn write_uniffi_module(
...
+                for binding_file in read_dir.flatten() {
+                    let target: PathBuf = base_path.join(binding_file.file_name());
+                    fs::copy(binding_file.path(), &target).context(format!(
```
Comment (messense) [inline src/module_writer.rs:1162]:
```suggestion
                    fs::copy(binding_file.path(), &target).with_context(|| format!(
```

**Comment by MatMaul** [comment]
Good idea, minimal test case added.

**Review by messense — APPROVED** [review]
Thanks!

**Pull request merged**